fcrepo-kernel/src/main/java/org/fcrepo/kernel/observer/FedoraEvent.java

```diff
@@ -119,13 +118,17 @@ public String getPath() throws RepositoryException {
      * from the end of property nodes).
      * @param e JCR Event
     **/
-    public static String getPath(final Event e) throws RepositoryException {
-        if (e.getType() == PROPERTY_ADDED   ||
-            e.getType() == PROPERTY_CHANGED ||
-            e.getType() == PROPERTY_REMOVED) {
-            return e.getPath().substring(0, e.getPath().lastIndexOf("/"));
+    public static String getPath(final Event e) {
+        try {
+            if (e.getType() == PROPERTY_ADDED   ||
```

Comment:
It would be better for this test to use a constant collection of `PROPERTY_ADDED`, `PROPERTY_CHANGED`, `PROPERTY_REMOVED` and `Collection.contains()`.